src/amo/components/VPNPromoBanner/index.js

```diff
@@ -0,0 +1,138 @@
+/* @flow */
+import config from 'config';
+import deepEqual from 'deep-eql';
+import * as React from 'react';
+import { compose } from 'redux';
+import { withRouter } from 'react-router-dom';
+
+import Button from 'amo/components/Button';
+import tracking from 'amo/tracking';
+import translate from 'amo/i18n/translate';
+import { makeQueryStringWithUTM } from 'amo/utils';
+import type { I18nType } from 'amo/types/i18n';
+import type { ReactRouterLocationType } from 'amo/types/router';
+
+import './styles.scss';
+import vpnLogo from './img/mozilla-vpn.svg';
+import dismissX from './img/x-close-black.svg';
+
+export const VPN_PROMO_CAMPAIGN = 'amo-vpn-promo';
+export const VPN_PROMO_CATEGORY = 'VPN Promo Banner';
+export const VPN_PROMO_CLICK_ACTION = 'vpn-promo-banner-click';
+export const VPN_PROMO_DISMISS_ACTION = 'vpn-promo-banner-dismiss';
+export const VPN_PROMO_IMPRESSION_ACTION = 'vpn-promo-banner-impression';
+export const VPN_URL = 'https://www.mozilla.org/products/vpn';
+
+export type Props = {||};
+
+export type DeafultProps = {|
+  _config: typeof config,
+  _tracking: typeof tracking,
+|};
+
+type InternalProps = {|
+  ...Props,
+  ...DeafultProps,
+  i18n: I18nType,
+  location: ReactRouterLocationType,
+|};
+
+export class VPNPromoBannerBase extends React.Component<InternalProps> {
+  static defaultProps: DeafultProps = {
+    _config: config,
+    _tracking: tracking,
+  };
+
+  onButtonClick: () => void = () => {
+    this.props._tracking.sendEvent({
+      action: VPN_PROMO_CLICK_ACTION,
+      category: VPN_PROMO_CATEGORY,
+    });
+  };
+
+  onDismiss: () => void = () => {
+    this.props._tracking.sendEvent({
+      action: VPN_PROMO_DISMISS_ACTION,
+      category: VPN_PROMO_CATEGORY,
+    });
+  };
+
+  onImpression: () => void = () => {
+    const { _config, _tracking } = this.props;
+
+    if (_config.get('enableFeatureVPNPromo')) {
+      _tracking.sendEvent({
+        action: VPN_PROMO_IMPRESSION_ACTION,
+        category: VPN_PROMO_CATEGORY,
+      });
+    }
+  };
+
+  componentDidMount() {
+    this.onImpression();
+  }
+
+  componentDidUpdate(prevProps: InternalProps) {
+    if (!deepEqual(this.props.location, prevProps.location)) {
+      this.onImpression();
+    }
+  }
+
+  render(): null | React.Node {
+    const { _config, i18n } = this.props;
+    const headline = i18n.gettext(
+      'Introductory offer ends soon: $4.99/month for Mozilla VPN',
+    );
+    const copy = i18n.gettext(
+      `Now's the time to protect your device against hackers and prying eyes.`,
+    );
+    const smallText = i18n.gettext(
+      'Offer only available in the United States, United Kingdom, Canada, New Zealand, Malaysia, and Singapore',
+    );
+
+    if (!_config.get('enableFeatureVPNPromo')) {
+      return null;
```

Comment:
Could be earlier or in the `Page` component.

Reply:
I chose to put it in this component to try to encapsulate the logic of the feature flag here. There's no reason the `Page` component even needs to know this flag exists, and in fact the only component that needs to know about it is *this* component.

As for moving it up, I wanted to keep the strings at the top of the render method, as they might need to be changed in the future, so I was treating them like `const`s that configure the component, and we usually put those at the top. Having said that, I can move it up to avoid the calls to `i18n.gettext`, as they will still be very near to the top of the method.